### ISSUE TITLE
Cannot stub `Array.prototype.sort`

### ISSUE THREAD

**Issue by artyhedgehog** [sinonjs/sinon#1796]
**Describe the bug**
Applying stub to `Array.prototype.sort` results in RangeError: Maximum call stack size exceeded.

**To Reproduce**
Run test like this:

```js
const sinon = require('sinon');

describe('Cannot stub Array.prototype.sort', () => {
  it('should not fall with call stack size exceeded error', () => {
    // Even making it always call through would not help:
    const stub = sinon.stub(Array.prototype, 'sort').callThrough();
    [1, 2, 3].sort();
    stub.restore();
  });
});
```

**Expected behavior**
Test should pass.

**Actual behavior**
Falls with error:
```
     RangeError: Maximum call stack size exceeded
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:1:1)
      at Array.proxy (node_modules/sinon/lib/sinon/spy.js:78:54)
      at Array.functionStub (node_modules/sinon/lib/sinon/stub.js:101:36)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:209:51)
      at Array.proxy (node_modules/sinon/lib/sinon/spy.js:78:54)
      at Array.functionStub (node_modules/sinon/lib/sinon/stub.js:101:36)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:209:51)
      at Array.proxy (node_modules/sinon/lib/sinon/spy.js:78:54)
      ...
```

**Context:**

* Library version: 5.0.7
* Environment: nodejs v8.11.1
* Command: /usr/bin/node ./node_modules/mocha/bin/_mocha sinon.spec.js | head -n20
* Other libraries you are using: mocha@5.1.1

**Additional context**
Looks similar to: https://github.com/sinonjs/sinon/issues/1521

**Comment by fatso83** [comment]
I am sure the fix to this is very similar to #1521. If you look grep the source for '.sort()' you should find it quickly. A PR is most welcome ;-)

**Comment by fatso83** [comment]
I think Phred fixed this in in #1826. Could you verify?

**Comment by fatso83** [comment]
Verified. Still present :(

**Comment by mroderick** [comment]
This was fixed in #1826, a test to verify it was added in #1876

**Comment by mroderick** [comment]
The fix for this has been published to NPM in `sinon@6.1.5`
